gpu/base/executor.cpp

```diff
@@ -46,34 +46,59 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 namespace gko {
 
 
-void CpuExecutor::raw_copy_to(const GpuExecutor *, size_type num_bytes,
+class device_guard {
+public:
+    device_guard(int device_id)
+    {
+        ASSERT_NO_CUDA_ERRORS(cudaGetDevice(&original_device_id));
+        ASSERT_NO_CUDA_ERRORS(cudaSetDevice(device_id));
+    }
+
+    ~device_guard() noexcept(false)
+    {
+        /* Ignore the error during stack unwinding for this call */
+        if (std::uncaught_exception())
+            cudaSetDevice(original_device_id);
+        else
+            ASSERT_NO_CUDA_ERRORS(cudaSetDevice(original_device_id));
```

Comment:
nit: always use curly braces with control flow statements. It avoids hard-to-catch bugs (say someone tries to add another statement to one of the branches, and forgets to add the braces).